Added Tween Sequence class

KoBeWi opened the pull request:
Hi, my first time contributing here, so hopefully I'm doing it correctly xd

This is inspired by https://github.com/godotengine/godot/issues/26529

Tween Sequence is a class that automatically creates Tweens for you and starts them. Basic usage looks like this:
```
var seq := TweenSequence.new(get_tree()).set_loops(1)
seq.append($Sprite, "position", Vector2(100, 100), 3).set_trans(Tween.TRANS_CUBIC).from_current()
seq.append_interval(1)
seq.append_callback($Sprite, "set_texture", [load("res://icon2.png")])
```
And produces this:
![R82UR2nGp1](https://user-images.githubusercontent.com/2223172/70644471-5e60a500-1c43-11ea-90c9-db663bb91092.gif)

The code isn't thoroughly tested though, so might not be ready to merge. Also taking any feedback whether something is lacking or something.

Xrayez commented on addons/godot-next/references/tween_sequence.gd:
```diff
@@ -0,0 +1,337 @@
+		tween.start()
+	
+	func __():
+		pass
```
I wonder what does this method? Implementation-specific?

KoBeWi replied:
Nothing. I could do interval with SceneTreeTimer, but since it's about Tweens, I did an empty callback tween that creates artificial interval. I just needed a method that does nothing.

willnationsdev commented:
This is very cool. It reminds me of the DOTween API that I just saw in the latest "Mario Galaxy Launcher" Mix and Jam video. I'll have to review this code later as I'm busy at the moment, but it sounds like a great contribution. :-)

KoBeWi commented:
> It reminds me of the DOTween API

This is what inspired me actually. I found Godot's Tweens lacking after using DOTween.

willnationsdev commented:
Would it be difficult to add a `.join` variation of method(s) then? Since, as I recall, it also had that feature. I can just merge this as is (after I review it), but wanted to ask if you wanted to do that too beforehand, if you haven't added that already.

KoBeWi commented:
It's already there, although I called it `parallel`. You can do `sequence.parallel().append*(...)` and the tweener will play together with the previous one. Although I didn't add a safeguard against doing it with empty sequence 🤔

Should I change the name to `join`, so it's `sequence.join().append*(...)`?

willnationsdev commented:
Nah. I don't really have a preference and I think parallel communicates the intent better anyway probably.

Xrayez commented:
You could add `join` as a wrapper to `parallel`, similar to how we have `append` as a wrapper to `push_back` in `Array`.

KoBeWi commented:
> You could add join as a wrapper to parallel, similar to how we have append as a wrapper to push_back in Array.

You mean "alias", right? I could do that.

KoBeWi commented:
So, I'm using the class in my project and keep finding things that can be improved, so I added some commits (see edits in previous comment). Here's changelog from the recent one:
- Better handling for invalid targets
- Starting a tween with freed object results in a warning instead of crash
- CallbackTweener now uses empty array for arguments by default
- Changed the way Tween starts to avoid multiple calls to `start()`

However, I've ran into a potential problem I'm not sure how to solve. When you free an object during tweening, the Tween doesn't let you know in any way that this happened. In case of TweenSequence it means pretty much a memory leak, because the Tween never finished and hangs infinitely (also further steps aren't executed). Any ideas how to avoid that?

willnationsdev commented:
Yeah, if you could squash these commits, that'd be great. Then I'll merge it all in. Sorry it's taken me so long to address this. And thank you so much for your contribution!

KoBeWi commented:
Squashed.

willnationsdev commented:
Thanks!
